File: Projek Android Studio/Online Image App/app/src/main/java/com/umar/onlineimageapp/MainActivity.java
package com.umar.onlineimageapp;

import android.os.Bundle;
import android.widget.ImageView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        // Initialize ImageView references
        ImageView imageView1 = findViewById(R.id.imageViewOnline1);
        ImageView imageView2 = findViewById(R.id.imageViewOnline2);

        // Define two publicly accessible image URLs (placeholders)
        String URL_GAMBAR_1 = "https://picsum.photos/800/600?random=1";
        String URL_GAMBAR_2 = "https://picsum.photos/800/600?random=2";

        // Load first image with placeholder
        com.squareup.picasso.Picasso.get()
                .load(URL_GAMBAR_1)
                .placeholder(R.drawable.placeholder_image)
                .into(imageView1);

        // Load second image without placeholder (direct load)
        com.squareup.picasso.Picasso.get()
                .load(URL_GAMBAR_2)
                .into(imageView2);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: Projek Android Studio/News App/app/src/main/java/com/umar/newsapp/MainActivity.kt
package com.umar.newsapp

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity() {

    private lateinit var drawerLayout: DrawerLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        // Apply window insets padding to root view (keeps previous behavior)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.drawer_layout)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        drawerLayout = findViewById(R.id.drawer_layout)
        val toolbar = findViewById<com.google.android.material.appbar.MaterialToolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val toggle = ActionBarDrawerToggle(
            this,
            drawerLayout,
            toolbar,
            R.string.drawer_open,
            R.string.drawer_close
        )
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        // Add fragment that hosts the WebView
        val fragment = NewsWebFragment()
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, fragment)
            .commit()

        val navView = findViewById<NavigationView>(R.id.navigation_view)
        navView.setNavigationItemSelectedListener { menuItem ->
            when (menuItem.itemId) {
                R.id.nav_bbc -> fragment.loadNewUrl("https://www.bbc.com/news")
                R.id.nav_cnn -> fragment.loadNewUrl("https://edition.cnn.com/")
                R.id.nav_aljazeera -> fragment.loadNewUrl("https://www.aljazeera.com/")
                R.id.nav_reuters -> fragment.loadNewUrl("https://www.reuters.com/")
                R.id.nav_guardian -> fragment.loadNewUrl("https://www.theguardian.com/international")
            }
            drawerLayout.closeDrawer(GravityCompat.START)
            true
        }
    }
}